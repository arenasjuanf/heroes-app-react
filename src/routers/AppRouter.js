import React from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
} from "react-router-dom";
import { LoginScreen } from '../components/login/LoginScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>

                <Route exact path="/login" component={ LoginScreen } />
                <Route exact path="/" component={  } />

               
                <Switch>
                  
                </Switch>
            </div>
        </Router>
    )
}
