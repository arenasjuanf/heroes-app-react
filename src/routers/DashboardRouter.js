import React from 'react'
import { Redirect, Route, Switch } from 'react-router'
import { Navbar } from '../components/ui/Navbar'
import { MarvelScreen } from '../components/marvel/MarvelScreen'
import { HeroeScreen } from '../components/heores/HeroScreen'
import { DcScreen } from '../components/dc/DcScreen' 

export const DashboardRouter = () => {
    return (
        <>
            <Navbar />

            <div>
                <Switch>
                    <Route exact path="/marvel" component={ MarvelScreen } />
                    <Route exact path="/heroe/:heroeId" component={ HeroeScreen } />
                    <Route exact path="/dc" component={ DcScreen } />
                    <Redirect to="/marvel" />

                </Switch>
            </div>

        </>
    )
}
