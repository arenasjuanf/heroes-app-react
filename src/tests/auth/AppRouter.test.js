import { mount } from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';
import { AppRouter } from '../../routers/AppRouter';

describe("Pruebas en <AppRouter />", () => {

    const contextValue = {
        dispatch: jest.fn(),
        user:{
            logged: false
        }
    }
    test("debe mostrar el login si no esoty autenticado", () => {

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        )
        
        expect(wrapper).toMatchSnapshot();
    })


    test("debe de mostrar el componente marvel si estÃ¡ autenticado", () => {

        const contextValue = {
            dispatch: jest.fn(),
            user:{
                logged: true,
                name:'Juan'
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <AppRouter />
            </AuthContext.Provider>
        )

        expect(wrapper.find('.navbar').exists()).toBe(true);
        // expect(wrapper).toMatchSnapshot();
    })
})