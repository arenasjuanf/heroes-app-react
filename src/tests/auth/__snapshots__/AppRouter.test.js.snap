// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe mostrar el login si no esoty autenticado 1`] = `
<div>
  <div
    className="container"
  >
    <h1>
      login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
  </div>
</div>
`;
